samples/Clue/array.clue

```diff
@@ -0,0 +1,10 @@
+local array = {1, 2, 3, 4}
+assert(array[1] == 1)
+
+table.insert(array, "hello")
+table.insert(array, "world")
+print(array[5], array[6])
```

Comment:
Please remove samples that aren't realistic examples of real-world usage. They're used to train Linguist's Bayesian classifier (which analyses keyword frequency to help classify ambiguous files correctly). From [`CONTRIBUTING.md`](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language):

> **"Hello world" examples will not be accepted.**

Reply:
are there any other samples which would not be considered realistic?

Reply:
Well, I was expecting to see some when I [searched for `*.clue` files](https://github.com/search?q=path%3A**%2F%3F*.clue&type=code), but from a cursory glance, I'm not sure if this language meets Linguist's popularity requirements (which are also documented in [`CONTRIBUTING.md`](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language), FYI).

However, judging in-the-world usage is @lildude's department (#5756), so I'll leave it to him to tell you whether or not this PR is worth pursuing at this time.

Reply:
Unfortunately, you are right on this one. There isn't much real world usage and I am aware it doesn't meet the requirements but I thought I should try anyways.

Reply:
Though it doesn't fulfil the usage criteria _now_, it may do so one day. Hence the [Pending Popularity](https://github.com/github/linguist/labels/Pending%20Popularity) label we use to keep track of languages that have yet to see sufficient currency on GitHub.